packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -395,6 +409,41 @@ export class Function extends FunctionBase {
     for (const event of props.events || []) {
       this.addEventSource(event);
     }
+
+    // Log retention
+    if (props.logRetentionDays) {
+      // Custom resource provider
+      const provider = new SingletonFunction(this, 'LogRetentionProvider', {
```

Comment:
Wrap this whole thing up into a construct that encapsulates all the custom resource details.

Reply:
OK, what about move this construct to `aws-cloudformation` or `aws-logs` (would cause a dep cycle) then? It could be used for CodeBuild and other services. For RDS, we have a similar situation when CloudWatch Logs are enabled (the log group is created during instance creation)

```ts
new LogRetention(this, 'LogRetention', {
  logGroupName: ...,
  retentionInDays: ...
});
```

Reply:
Or to a new package (`@aws-cdk/custom-resource`) as discussed in #1850.

Reply:
But this will create dependency cycles (`SingletonFunction`)...